Sources/ClientRuntime/Networking/Http/SdkHttpRequest.swift

```diff
@@ -83,6 +89,28 @@ extension SdkHttpRequest {
         httpRequest.body = nil
         return httpRequest
     }
+
+    public func toURLRequest() async throws -> URLRequest {
+        // Set URL
+        guard let url = self.endpoint.url else {
+            throw ClientError.dataNotFound("Failed to construct URLRequest due to missing URL.")
+        }
+        var request = URLRequest(url: url)
+        // Set method type
+        request.httpMethod = self.method.rawValue
+        // Set body
+        guard let body = try await self.body.readData() else {
+            throw ClientError.serializationFailed("Failed to construct URLRequest due to HTTP body conversion failure.")
+        }
+        request.httpBody = body
```

Comment:
```
// Set body, handling any serialization errors
do {
    request.httpBody = try await self.body.readData()
} catch {
    throw ClientError.serializationFailed("Failed to construct URLRequest due to HTTP body conversion failure.")
}
```

In this case do / catch is more idiomatic than guard. From my understanding guard is meant to be used when we have a clear & simple condition we need to in order to proceed such as "X things is true" or "Y exists". In this case we are executing a method that we know may throw an error so its more of a complex statement. Therefore it is easier to read using do/catch as we're saying "we're gonna execute this block of code **knowing** that it may fail" vs saying "we don't expect this statement to fail but we need to **ensure** it never does to continue on a happy path"

Reply:
Changed as suggested.